### PR TITLE
Testing To Do list: part 1

### PR DESCRIPTION
**Pull request by Xghayor**
In this we completed:
1.Add test to add item function
2.Add test to delete item function.

### CODE REVIEW

File: src/modules/test/deletetask.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  test('delete exactly one <li> element', () => {
+    deleteTask(listItem, tasks);
+
+    const liElements = taskList.querySelectorAll('li');
+
+    expect(liElements.length).toBe(0);
+    expect(updateTaskIndexesInStorage).toHaveBeenCalledWith(tasks);
+    expect(updateLocalStorage).toHaveBeenCalledWith(tasks);
+  });
```
Comment (usorfaitheloho) [inline src/modules/test/deletetask.test.js:38]: You did well with the `test implementation` for your TODO list, kindly note that you are to write tests for both the `add item` and `delete item`, while you did well writing tests for the deleteTask, you should also write for the addTask so as to fully check if they are performing what is expected of them 😄 

image

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @Xghayor  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * Used jest framework for testing ✔️
> * Descriptive pull request ✔️
> * Used GitHub flow ✔️
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/test/deletetask.test.js (hunk 1 above)
Comment (Xghayor) [reply src/modules/test/deletetask.test.js:38]: @usorfaitheloho 
Hi, Did you even bother to check the test folder?
when you run npx jest, how many files it check for test?

**Review by V-Blaze — APPROVED** [review]
Hi @Xghayor, @ethred, 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Wrote test for add and delete functionality 💯
-  All tests are passing👍
-  Descriptive pull request title🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/test/deletetask.test.js (hunk 1 above)
Comment (V-Blaze) [reply src/modules/test/deletetask.test.js:38]: @Xghayor I apologize for the misunderstanding from the previous reviewer.

While mistakes can happen, If there was an error in the code review, it's important to address it constructively. We should all aim to communicate respectfully and professionally, emphasizing that collaboration and learning from mistakes are part of the development process. :+1: :100: :wave:

**Pull request merged**